### PR TITLE
refactor(components): [rate] use css var & disable to disabled & star-color to fill-color

### PR DESCRIPTION
**Pull request by YunYouJun**
- use css vars for rate & remove computed inline style
- rename partial `disable` to `disabled` to unify
- rename `--el-rate-star-color` to `--el-rate-fill-color`

---

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/rate/__tests__/rate.vitest.tsx [hunk 1]
```diff
@@ -83,10 +83,10 @@ describe('Rate.vue', () => {
       <Rate v-model={value.value} colors={['#99A9BF', '#F7BA2A', '#FF9900']} />
     ))
 
-    const thirdStar = (
-      wrapper.findAll('.el-rate__item')[2].element as HTMLElement
-    ).querySelector<HTMLElement>('.el-rate__icon')!
-    expect(thirdStar.style.color).toMatchInlineSnapshot('"rgb(255, 153, 0)"')
+    const rateEl = wrapper.find('.el-rate').element as HTMLElement
+    expect(rateEl.style.getPropertyValue('--el-rate-fill-color')).toBe(
```
Comment (sxzz) [inline packages/components/rate/__tests__/rate.vitest.tsx:87]: How about using `toMatchInlineSnapshot`

**Review by sxzz — APPROVED** [review]

File: packages/components/rate/__tests__/rate.vitest.tsx (hunk 1 above)
Comment (YunYouJun) [reply packages/components/rate/__tests__/rate.vitest.tsx:87]: Maybe it's simpler for using `toBe` directly.

> It’s typically better to make more specific assertions than to use snapshots. These kinds of tests include implementation details so they break easily, and teams can get desensitized to snapshot breakages. Selectively [mocking some child components](https://reactjs.org/docs/testing-recipes.html#mocking-modules) can help reduce the size of snapshots and keep them readable for the code review.

> https://reactjs.org/docs/testing-recipes.html#snapshot-testing

File: packages/components/rate/src/rate.vue [hunk 2]
```diff
@@ -1,26 +1,29 @@
 <template>
   <div
-    :class="rateClasses"
+    :class="[rateClasses, rateDisabled && ns.is('disabled')]"
```
Comment (JeremyWuuuuu) [inline packages/components/rate/src/rate.vue:3]:
```suggestion
    :class="[rateClasses, ns.is('disabled', rateDisabled)]"
```

File: packages/components/rate/src/rate.vue [hunk 3]
```diff
@@ -1,26 +1,29 @@
...
       @mouseleave="resetCurrentValue"
       @click="selectValue(item)"
     >
       <el-icon
-        :class="[ns.e('icon'), { hover: hoverIndex === item }]"
-        :style="getIconStyle(item)"
+        :class="[
+          ns.e('icon'),
+          { hover: hoverIndex === item },
+          item <= currentValue && ns.is('active'),
```
Comment (JeremyWuuuuu) [inline packages/components/rate/src/rate.vue:25]:
```suggestion
         ns.is('active',  item <= currentValue),
```

**Review by jw-foss — APPROVED** [review]

**Pull request merged**